Sources/Exporters/DatadogExporter/Persistence/FileReader.swift

```diff
@@ -62,6 +62,21 @@ internal final class FileReader {
         return nil
     }
 
+    func onRemainingBatches(process: (Batch)->()) -> Bool {
+        queue.sync {
+            do {
+                try orchestrator.getAllFiles(excludingFilesNamed: Set(filesRead.map { $0.name }))?.forEach {
```

Comment:
so this will take the list of files once and any new file will be ignored, correct? Nothing prevents code from writing new ones while flushing, right?

Reply:
just making sure there is no chance flush will never end keep reading and producing files...

Reply:
perhaps say it explicitly

Reply:
Writer should not be working when this code is run, but anyway the getAllFIles() method gets all the remaining file at once, so if any other file appeared after that read while the previous are being sent it wont be read. 
But as you said much better having this info as a comments.